### PR TITLE
[Websocket] Path Param Support

### PR DESCRIPTION
**Pull request by irunika**
This PR introduces path param support for WebSocket in Ballerina.
_Note: Please merge https://github.com/ballerinalang/ballerina/pull/4043 before merging this PR._

This Resolves https://github.com/ballerinalang/ballerina/issues/3845

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/WebSocketServicesRegistry.java [hunk 1]
```diff
@@ -71,81 +79,112 @@ public void registerService(WebSocketService service) {
...
+    }
+
+    private void deployServiceEndpoints(Set<WebSocketService> deployedSlaveServiceSet) {
+        serviceEndpoints.entrySet().forEach(serviceEntry -> {
+            WebSocketService service = serviceEntry.getValue();
+            Annotation configAnnotation =
+                    service.getAnnotation(Constants.WEBSOCKET_PACKAGE_NAME, Constants.ANNOTATION_CONFIGURATION);
+            if (configAnnotation == null && !deployedSlaveServiceSet.contains(service)) {
+                String errorMsg =
+                        String.format("Service %s is without configuration annotation and also not referred " +
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/WebSocketServicesRegistry.java]: Isn't is "Service %s is deployed without ..."?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/WebSocketServicesRegistry.java [hunk 2]
```diff
@@ -71,81 +79,112 @@ public void registerService(WebSocketService service) {
...
+                                              "in HTTP service cannot be deployed.", service.getName());
+                throw new BallerinaConnectorException(errorMsg);
+            }
+            String basePath = findFullWebSocketUpgradePath(service);
+            if (basePath == null) {
+                if (deployedSlaveServiceSet.contains(service)) {
+                    return;
                 }
+                String errorMsg =
+                        String.format("Service %s is without base path in configuration annotation and also " +
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/WebSocketServicesRegistry.java]: Isn't is "Service %s is deployed without ..."?

**Comment by irunika** [comment]
This PR is being updated. Need to wait until issue https://github.com/ballerinalang/ballerina/issues/4146 is fixed.

**Comment by irunika** [comment]
Need to resolve https://github.com/ballerinalang/ballerina/issues/4146 before merging this PR and also by making this changes this will have many changes. Will send another PR with new changes.

**Pull request closed without merging**